production/docker-compose.yml

```diff
@@ -0,0 +1,51 @@
+version: '3.4'
+
+services:
+  nodepki:
+    image: pschmid/nodepki-docker:dev
+    restart: always
+    env_file:
+      - .env
+    environment:
+      TZ: Europe/Zurich
+    healthcheck:
+      test: ["CMD-SHELL", "curl --fail http://localhost:5000 || exit 1"]
+      interval: 30s
+      timeout: 10s
+      retries: 3
+    volumes:
+      - /opt/data/nodepki:/opt/nodepki/data
+      - /opt/data/nodepki-client:/opt/nodepki-client/data
+      - /opt/data/nodepki-webclient:/opt/nodepki-webclient/data
+        # TODO: Check: Is this last Docker Volume really required?
+      - /opt/data/nodepki-certs:/opt/nodepki-client/out
+    extra_hosts:
+      - ${CA_API_SERVER_URL}:127.0.0.1
+    networks:
+      - traefik_transit
+    labels:
+      traefik.ca-admin.port: 5000
+      traefik.ca-admin.frontend.rule: Host:${CA_WEBCLIENT_SERVER_URL}
+      traefik.ca-admin.protocol: http
+      traefik.ca.port: ${CA_API_SERVER_PLAIN_PORT}
+      traefik.ca.frontend.rule: Host:${CA_API_SERVER_URL}
+      traefik.ca.protocol: http
+      traefik.ocsp.port: ${CA_OSCP_SERVER_PORT}
+      traefik.ocsp.frontend.rule: Host:${CA_OSCP_SERVER_URL}
+      traefik.ocsp.protocol: http
+
+  traefik:
+    image: traefik:1.6-alpine
+    restart: always
+    ports:
```

Comment:
Pick up from .env?

Reply:
What value of these 4 lines would you like to take from the `.env` file?

Reply:
I didn't know github show the 4 lines above the selected line. The interesting lines are below the 'ports' line since the ports are defined in the .env file those could be picked up here.

Reply:
Ah, you mean:
```bash
    ports:
      - 80:80
      - 443:443
      - 8080:8080
```

Well, I don't think we can use the values from `.env` since these 3 Traefik port definitions are not directly related to the Nodepki used ports. These Traefik ports are just for **any** incoming HTTP (`80:80`), HTTPS ( `443:443`) and Traefik WebUI traffic (`8080:8080`). Traefik then routes the traffic to the corresponding Docker service based on the URI inside the request and the set - on the service - `traefik.ca-admin.frontend.rule` Docker label.

Regards,
Philip